### ISSUE TITLE
Fixes for LT02 not applied for case statements in update statements in BigQuery transactions

### ISSUE THREAD

**Issue by dbrakman** [sqlfluff/sqlfluff#6274]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Case statements in update statements in BigQuery transactions prevent applying fixes that include rule LT02.

### Expected Behaviour

The minimal query provided under the "How to reproduce" section passes validation with no fixes or errors.

### Observed Behaviour

Fixes for LT02 cannot be applied, with the following error:
```
==== sqlfluff ====
sqlfluff:                2.3.5 python:                 3.8.19
implementation:        cpython verbosity:                   1
dialect:              bigquery templater:               jinja
rules:                               LT02
==== finding fixable violations ====
WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning. 
== [nbdist/mwe.sql] FAIL                                                                                                
L:   1 | P:   6 | LT02 | Expected line break and indent of 4 spaces before
                       | 'transaction'. [layout.indent]
L:   5 | P:  41 | LT02 | Expected line break and no indent before 'end'.
                       | [layout.indent]
  [2 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [2 unfixable linting violations found]
```

### How to reproduce

Save the following as `mwe.sql`:
```sql
begin transaction;

update myproject.mydataset.mytable
set
    b = case when a then false else true end
where
    true

commit transaction;
```

Save the Configuration as .sqlfluff, then run the command `sqlfluff fix -v mwe.sql` .

### Dialect

bigquery

### Version

2.3.5

### Configuration

```
[sqlfluff]
dialect = bigquery
rules = LT02
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by dbrakman** [comment]
One workaround is to wrap the case statement in parentheses. The following query exhibits the expected behavior in version 2.3.5.
```sql
begin transaction;

update myproject.mydataset.mytable tgt
set
    b = (case when a then false else true end)
where
    true;

commit transaction;
```

I'm not sure which specific version resolved the issue, but version 3.2.0 exhibits the expected behavior on the original minimal query.
